app/uk/gov/hmrc/agentaccesscontrol/services/AuthorisationService.scala

```diff
@@ -77,17 +78,12 @@ class AuthorisationService @Inject()(
                       result = false,
                       AccessResponse.toReason(accessResponse))
         Future.successful(accessResponse)
-      case _ =>
```

Comment:
review whether this case could be hit

Reply:
authDetails match {
    case nonMtdAgentAuthDetails @ AuthDetails(Some(saAgentReference), _, _, _, _) =>
    case mtdAgentAuthDetails @ AuthDetails(_, Some(arn), _, _, _) =>
    case agentAuthDetails @ AuthDetails(None, _, _, _, _) =>
    case _ =>
}
The 1st and 3rd case cover all potential cases as they test for Some and None on the first item and don't care about anything else so the _ case is unreachable